app/controllers/projects/queries_controller.rb

```diff
@@ -71,15 +72,27 @@ def update
     render_result(call, success_i18n_key: "lists.update.success", error_i18n_key: "lists.update.failure")
   end
 
-  def toggle_public
-    to_be_public = !@query.public?
+  def toggle_public # rubocop:disable Metrics/AbcSize
+    to_be_public = ActiveRecord::Type::Boolean.new.cast(params["value"])
     i18n_key = to_be_public ? "lists.publish" : "lists.unpublish"
 
     call = Queries::Projects::ProjectQueries::PublishService
              .new(user: current_user, model: @query)
              .call(public: to_be_public)
 
-    render_result(call, success_i18n_key: "#{i18n_key}.success", error_i18n_key: "#{i18n_key}.failure")
+    respond_to do |format|
+      format.turbo_stream do
+        # Load shares and replace the modal
+        strategy = SharingStrategies::ProjectQueryStrategy.new(@query, user: current_user)
+        shares = strategy.shares_query({}).results
+        replace_via_turbo_stream(component: Shares::ModalBodyComponent.new(strategy:, shares:, errors: []))
```

Comment:
Maybe better to pass only `strategy` to `ModalBodyComponent`, to not pass `shares` separately after getting it through `strategy method`

Reply:
Yes, I will do exactly that in the next PR because I will need to dynamically add something to the shares